src/components/booksList/BooksList.tsx

```diff
@@ -0,0 +1,104 @@
+import { FC } from "react";
+import "./BooksList.css";
+import { IBook } from "../../interfaces";
+import Book from "../book/Book";
+import Pagination from "../pagination/Pagination";
+import { useDispatch, useSelector } from "react-redux";
+import { unselectAllBooks } from "../../store/slices/selectedBooksSlice";
+import {
+  selectBook,
+  unselectBook,
+} from "../../store/slices/selectedBooksSlice";
+import { RootState } from "../../store/store";
+
+type Props = {
+  totalPages: number;
+  currentPage: number;
+};
+
+const BooksList: FC<Props> = ({ totalPages, currentPage }) => {
+  const list = useSelector(
+    (state: RootState) => state.currentPageItems.currentPageItems,
+  );
+  const selectedBooks = useSelector(
+    (state: RootState) => state.selectedBooks.selectedBooks,
+  );
+
+  const dispatch = useDispatch();
+
+  const handleClick = (book: IBook, e: React.ChangeEvent<HTMLInputElement>) => {
+    if ((e.target as HTMLInputElement).checked) {
```

Comment:
```suggestion
    if (e.target.checked) {
```
ты строчкой выше описала тип, теперь и так работает